### DISCUSSION TITLE
Is there a way to extend data class with additional properties?

### DISCUSSION THREAD

**Discussion by msnegurski** [spatie/laravel-data#204] [Q&A] [answered]
We have several pretty big data classes (>50 properties) and they differ only by several properties. It would be really cool if there was a way to create base data class for them and just extend it with several properties, instead of having two almost identical classes, but I have no idea how to do it.

**Comment by rubenvanassche** [comment, marked as answer]
This should work, added a test to check it: https://github.com/spatie/laravel-data/blob/0e8750426fe0e07e27b5f21051c3e8735fa58bcd/tests/DataTest.php#L2376-L2387

**Reply by msnegurski** [reply]
Thank you! WIll try it in next few days

**Reply by msnegurski** [reply]
Alright, that totally works, thank you.
